src/Tools/dotnet-monitor/Extensibility/OutputParser.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.Logging;
+using System;
+using System.Diagnostics;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Extensibility
+{
+    internal class OutputParser<TResult> : IDisposable where TResult : class
```

Comment:
Is this going to be used in other places? If not, inline it with the ProgramExtension class.